### PR TITLE
#586: Delegate the construction of the map of arguments to a private static method to be compliant with ConstructorOnlyInitializesOrCallOtherConstructors

### PR DESCRIPTION
**Pull request by essobedo**
Fix for https://github.com/yegor256/takes/issues/586

The goal of this PR is to make the class Options compliant with the rule  ConstructorOnlyInitializesOrCallOtherConstructors by delegating the construction of the map of arguments to a private static method.

### CODE REVIEW

**Comment by davvd** [comment]
@essobedo Let me find a reviewer for this pull request, thanks for submitting it

**Comment by davvd** [comment]
@longtimeago review this one, please

File: src/main/java/org/takes/http/Options.java [hunk 1]
```diff
@@ -57,7 +57,11 @@
     private final transient ConcurrentMap<String, String> map;
 
     /**
-     * Ctor.
+     * Constructs an {@code Options} with the specified arguments. The expected
```
Comment (longtimeago) [inline src/main/java/org/takes/http/Options.java]: @essobedo Great description, but this particular case the constructor is package-protected, not a part of API
Let's leave it as is, but, usually, it's too much :)

File: src/main/java/org/takes/http/Options.java [hunk 2]
```diff
@@ -66,32 +70,15 @@
     }
 
     /**
-     * Ctor.
+     * Constructs an {@code Options} with the specified arguments. The expected
```
Comment (longtimeago) [inline src/main/java/org/takes/http/Options.java]: @essobedo same here

File: src/main/java/org/takes/http/Options.java [hunk 3]
```diff
@@ -199,4 +186,38 @@ public long maxLatency() {
         }
         return msec;
     }
+
+    /**
+     * Convert the provided arguments into a Map. The expected
```
Comment (longtimeago) [inline src/main/java/org/takes/http/Options.java]: @essobedo this is really verbose. You've repeated implementation almost fully. Code should be the only source of true :)

File: src/main/java/org/takes/http/Options.java [hunk 4]
```diff
@@ -199,4 +186,38 @@ public long maxLatency() {
...
+     * empty {@code String} will be assigned to that parameter by default.
+     * @param args Arguments to parse.
+     * @return Map A map containing all the arguments which are actually
+     *  key/value pairs.
+     * @throws IllegalStateException If an argument doesn't match with the
+     *  expected format.
+     */
+    private static ConcurrentMap<String, String> asMap(
+        final Iterable<String> args) throws IllegalStateException {
+        final ConcurrentMap<String, String> map =
```
Comment (longtimeago) [inline src/main/java/org/takes/http/Options.java]: @essobedo field type can be relaxed to simple `Map` as we don't use `ConcurrentMap` specific operations

**Comment by longtimeago** [comment]
@essobedo thanks! Please, see several remarks above

**Comment by longtimeago** [comment]
@essobedo much better :) Thanks!

**Comment by longtimeago** [comment]
@rultor merge

**Comment by essobedo** [comment]
@longtimeago thx for the review here is a less verbose version of the PR. I globally removed what I added as Javadoc following "Code should be the only source of true" principle

**Comment by essobedo** [comment]
@yegor256 please confirm

**Comment by essobedo** [comment]
@longtimeago PR updated to fix the conflicts, please review

**Comment by longtimeago** [comment]
@rultor merge

**Comment by essobedo** [comment]
@yegor256 This PR is waiting for a confirmation for more than a week now, so could you please do something about it? confirm it or reject it but please do something about it.

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by essobedo** [comment]
@yegor256 Thank you very much

**Comment by longtimeago** [comment]
@yegor256 Looks like the CR payment is missed here

**Comment by longtimeago** [comment]
@yegor256 ping :)

**Comment by longtimeago** [comment]
@yegor256 payment is missing for this ticket. Please, check

**Comment by longtimeago** [comment]
@rultor deploy

**Comment by yegor256** [comment]
@longtimeago give it a few more days, will be paid

**Comment by yegor256** [comment]
@rultor deploy

**Comment by davvd** [comment]
@ypshenychka please, let us know what do you think about this ticket, according to our [QA rules](http://at.teamed.io/qa.html)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by ypshenychka** [comment]
@davvd Quality is good.

**Comment by davvd** [comment]
> @davvd Quality is good.

@ypshenychka thanks a lot

**Comment by davvd** [comment]
@longtimeago just added 10 mins to @ypshenychka (for QA), payment ID is `79175563`. Much appreciated! **27 mins** was added to your account, payment ID `79175579`, time spent is 397 hours and 57 mins. review comments (c=12) added as a bonus. +27 to your rating, your total score is [+6574](http://www.netbout.com/b/35024?open=rating)

**Pull request merged**